examples/linear_model/plot_poisson_regression_non_normal_loss.py

```diff
@@ -0,0 +1,455 @@
+# Let's load the motor claim dataset. We ignore the severity data for this
+# study for the sake of simplicitly.
+#
+# We also subsample the data for the sake of computational cost and running
+# time. Using the full dataset would lead to similar conclusions.
+
+df = load_mtpl2(n_samples=300000)
+
+# Correct for unreasonable observations (that might be data error)
+df["Exposure"] = df["Exposure"].clip(upper=1)
+
+##############################################################################
+# The remaining columns can be used to predict the frequency of claim events.
+# Those columns are very heterogeneous with a mix of categorical and numeric
+# variables with different scales, possibly very unevenly distributed.
+#
+# In order to fit linear models with those predictors it is therefore
+# necessary to perform standard feature transformations as follows:
+
+log_scale_transformer = make_pipeline(
+    FunctionTransformer(np.log, validate=False),
+    StandardScaler()
+)
+
+linear_model_preprocessor = ColumnTransformer(
+    [
+        ("passthrough_numeric", "passthrough",
+            ["BonusMalus"]),
+        ("binned_numeric", KBinsDiscretizer(n_bins=10),
+            ["VehAge", "DrivAge"]),
+        ("log_scaled_numeric", log_scale_transformer,
+            ["Density"]),
+        ("onehot_categorical", OneHotEncoder(),
+            ["VehBrand", "VehPower", "VehGas", "Region", "Area"]),
+    ],
+    remainder="drop",
+)
+
+##############################################################################
+# The number of claims (``ClaimNb``) is a positive integer that can be modeled
+# as a Poisson distribution. It is then assumed to be the number of discrete
+# events occurring with a constant rate in a given time interval
+# (``Exposure``, in units of years). Here we model the frequency
+# ``y = ClaimNb / Exposure``, which is still a (scaled) Poisson distribution,
+# and use ``Exposure`` as ``sample_weight``.
+
+df["Frequency"] = df["ClaimNb"] / df["Exposure"]
+
+print(
+   pd.cut(df["Frequency"], [-1e-6, 1e-6, 1, 2, 3, 4, 5]).value_counts()
+)
+
+print("Average Frequency = {}"
+      .format(np.average(df["Frequency"], weights=df["Exposure"])))
+
+print("Percentage of zero claims = {0:%}"
+      .format(df.loc[df["ClaimNb"] == 0, "Exposure"].sum() /
+              df["Exposure"].sum()))
+
+##############################################################################
+# It is worth noting that 92 % of policyholders have zero claims, and if we
+# were to convert this problem into a binary classification task, it would be
+# significantly imbalanced.
+#
+# To evaluate the pertinence of the used metrics, we will consider as a
+# baseline a "dummy" estimator that constantly predicts the mean frequency of
+# the training sample.
+
+df_train, df_test = train_test_split(df, random_state=0)
+
+dummy = make_pipeline(
+    linear_model_preprocessor,
+    DummyRegressor(strategy='mean')
+)
+dummy.fit(df_train, df_train["Frequency"],
```

Comment:
The target and the exposure are all in `df_train`. This is currently okay because the column transformer drops them. Do we want to promote this kind of usage?

Reply:
We could define,
```py
X_train = df_train.drop(['ClaimNB', 'Frequency', 'Exposure'], axis=1)
y_train = df_train['Frequency']
```
to be safe, but I think relying on ColumnTransformer to drop the rest might be OK.  Particularly with `remainder='drop'` being the default.   Also it's more explicit about the column which is the target (in the second example there are multiple targets), while `fit(X_train, df_train['Frequency'])` also looks not very common to me.

In the second tweedie example, we feed this directly to column transformer, and I definitely find,
```py
ct = ColumnTransformer(..., remainder='drop')
ct.fit(df_train.drop(['ClaimNB', "ClaimAmount", 'Frequency', 'Exposure'], axis=1))
```
to be redundant.

Reply:
Just a remark: On top of having `Exposure` as `sample_weight`, it is possible to put it into the feature matrix `X` as well. For example, policies that are only valid a part of a year might have higher (or lower) claims frequency. To keep it simple, I would not do so the example.